examples/solids/elasticity.c

```diff
@@ -140,14 +134,26 @@ int main(int argc, char **argv) {
 
   // -- Create distributed DM from mesh file
   ierr = CreateDistributedDM(comm, appCtx, &dmOrig); CHKERRQ(ierr);
+  VecType vectype;
+  switch (memTypeBackend) {
+  case CEED_MEM_HOST: vectype = VECSTANDARD; break;
+  case CEED_MEM_DEVICE: {
+    const char *resolved;
+    CeedGetResource(ceed, &resolved);
+    if (strstr(resolved, "/gpu/cuda")) vectype = VECCUDA;
+    else if (strstr(resolved, "/gpu/hip")) vectype = VECHIP;
```

Comment:
Do we need to add the OCCA bypass here as well?

Reply:
It's currently screened out in `tap.sh` and `junit.py`. I presume that #678 can be fixed before OCCA adds support for VLA (and thus is usable for our solids and fluids examples).

Reply:
That's fair. 

One of these days we need to refactor all of these helper functions into a single place so that we don't have to keep track of these sorts of differences between examples.

Reply:
Yes, I actually want that to go into PETSc, which has enough context (when used in a mode that would call libCEED) to make such decisions.